src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/References/AbstractReferenceHandler.cs

```diff
@@ -108,33 +109,28 @@ private static async Task<bool> GetAttributeTreatAsUsedAsync(IProjectProperties
         {
             Dictionary<string, string> propertyValues = new ();
 
-            IProjectItem items = await GetProjectItems(selectedConfiguredProject, itemSpecification);
+            IProjectItem items = await GetProjectItemsAsync(selectedConfiguredProject, itemSpecification);
 
             var propertyNames = await items.Metadata.GetPropertyNamesAsync();
 
             foreach (var property in propertyNames)
             {
                 var value = await items.Metadata.GetEvaluatedPropertyValueAsync(property);
-                propertyValues.Add(string.Copy(property), string.Copy(value));
+                propertyValues.Add(property, value);
             }
 
             return propertyValues;
         }
 
-        public async Task SetAttributes(ConfiguredProject selectedConfiguredProject, string itemSpecification, Dictionary<string, string>? projectPropertiesValues)
+        public async Task SetAttributesAsync(ConfiguredProject selectedConfiguredProject, string itemSpecification, Dictionary<string, string> projectPropertiesValues)
         {
-            if (projectPropertiesValues is null)
-            {
-                return;
-            }
-
-            IProjectItem items = await GetProjectItems(selectedConfiguredProject, itemSpecification);
+            IProjectItem items = await GetProjectItemsAsync(selectedConfiguredProject, itemSpecification);
 
             if (items != null)
```

Comment:
If this can return `null`, it should be annotated (both the local variable and the `GetProjectItemsAsync` method).